### PR TITLE
Refactor: Indexer Performance Improvements

### PR DESCRIPTION
**Pull request by gregfromstl**
Refactors the Indexer backfilling process for faster/bulk database insertions and parallel log fetching. No new features are added, this is purely a refactor and the logs not yet added to the database still need to be implemented.

I believe this completes #8

### CODE REVIEW

File: lib/eth/src/id_registry.rs [hunk 1]
```diff
@@ -124,17 +122,67 @@ impl<T: JsonRpcClient + Clone> Contract<T> {
...
 
-        let fid_row = db::FidRow {
-            fid: parsed_log.id.as_u64() as i64,
-            registered_at: timestamp.into(),
-            chain_event_id: event_row.id,
-            custody_address: parsed_log.to.to_fixed_bytes(),
-            recovery_address: parsed_log.recovery.to_fixed_bytes(),
-        };
+        Ok((
+            db::FidRow {
```
Comment (haardikk21) [inline lib/eth/src/id_registry.rs:127]: minor style nitpick: assigning `ChainEventRow` to a var but creating the `FidRow` inline 

not that it matters tbh and i'd rather functionality over style at this point, but just pointing it out for now

dont need to fix for merge - not a blocker

File: lib/eth/src/indexer.rs [hunk 2]
```diff
@@ -284,55 +356,83 @@ impl<T: JsonRpcClient + Clone> Indexer<T> {
...
+        let start_time = std::time::Instant::now();
 
         // fetch logs in range [current_block_num, current_block_num + 2000]
         while current_block <= end_block {
-            let percent_complete =
-                (current_block - start_block) as f64 / (end_block - start_block) as f64;
-            log::info!("events sync progress = {:.2}%", percent_complete * 100.0);
-
-            // Clear block timestamp cache to avoid overloading it with useless data
-            self.block_timestamp_cache.clear();
```
Comment (haardikk21) [inline lib/eth/src/indexer.rs:301]: why was this removed? feels like unnecessary memory overhead to keep the cache there given values in the cache will not be accessed once we move on to another "chunk"

File: lib/eth/src/id_registry.rs (hunk 1 above)
Comment (gregfromstl) [reply lib/eth/src/id_registry.rs]: no worries, easy fix

File: lib/eth/src/indexer.rs (hunk 2 above)
Comment (gregfromstl) [reply lib/eth/src/indexer.rs:301]: Good catch, this shouldn't have been removed

**Review by haardikk21 — APPROVED** [review]
LGTM!

**Comment by gregfromstl** [comment]
Sweet could you merge? I don't have permissions
